contracts/contracts/strategies/AaveStrategy.sol

```diff
@@ -0,0 +1,202 @@
+
+        IERC20(_asset).safeTransfer(_recipient, amountWithdrawn);
+
+        emit Withdrawal(_asset, address(aToken), amountWithdrawn);
+    }
+
+    /**
+     * @dev Remove all assets from platform and send them to Vault contract.
+     */
+    function liquidate() external onlyVaultOrGovernor {
+        for (uint256 i = 0; i < assetsMapped.length; i++) {
+            // Redeem entire balance of cToken
+            IAaveAToken aToken = _getATokenFor(assetsMapped[i]);
+            uint256 balance = aToken.balanceOf(address(this));
+            if (balance > 0) {
+                aToken.redeem(balance);
+                // Transfer entire balance to Vault
+                IERC20 asset = IERC20(assetsMapped[i]);
+                asset.safeTransfer(
+                    vaultAddress,
+                    asset.balanceOf(address(this))
+                );
+            }
+        }
+    }
+
+    /**
+     * @dev Get the total asset value held in the platform
+     * @param _asset      Address of the asset
+     * @return balance    Total value of the asset in the platform
+     */
+    function checkBalance(address _asset)
+        external
+        view
+        returns (uint256 balance)
+    {
+        // Balance is always with token cToken decimals
+        IAaveAToken aToken = _getATokenFor(_asset);
+        balance = aToken.balanceOf(address(this));
+    }
+
+    /**
+     * @dev Retuns bool indicating whether asset is supported by strategy
+     * @param _asset Address of the asset
+     */
+    function supportsAsset(address _asset) external view returns (bool) {
+        return assetToPToken[_asset] != address(0);
+    }
+
+    /**
+     * @dev Approve the spending of all assets by their corresponding cToken,
+     *      if for some reason is it necessary.
+     */
+    function safeApproveAllTokens() external onlyGovernor {
+      uint256 assetCount = assetsMapped.length;
+      address lendingPoolVault = _getLendingPoolCore();
+      // approve the pool to spend the bAsset
+      for(uint i = 0; i < assetCount; i++){
+        address asset = assetsMapped[i];
+        // Safe approval
+        IERC20(asset).safeApprove(lendingPoolVault, 0);
+        IERC20(asset).safeApprove(lendingPoolVault, uint256(-1));
+      }
+    }
+
+    /**
+     * @dev Internal method to respond to the addition of new asset / cTokens
+     *      We need to approve the cToken and give it permission to spend the asset
+     * @param _asset Address of the asset to approve
+     * @param _aToken This aToken has the approval approval
+     */
+    function _abstractSetPToken(address _asset, address _aToken) internal {
+      address lendingPoolVault = _getLendingPoolCore();
+      IERC20(_asset).safeApprove(lendingPoolVault, 0);
+      IERC20(_asset).safeApprove(lendingPoolVault, uint256(-1));
+    }
+
+    /**
+     * @dev Get the weighted APR for all assets in strategy.
+     * @return APR in 1e18
+     */
+    function getAPR() external view returns (uint256) {
```

Comment:
We can remove all the APR methods now, they are gone from the abstraction too

Reply:
These methods have not been removed yet. @crazybuster @franckc